addons/website/static/src/snippets/s_website_form/000.js

```diff
@@ -236,6 +236,11 @@ odoo.define('website.s_website_form', function (require) {
                 if (initialValue) {
                     fieldEl.setAttribute('value', initialValue);
                 } else {
+                    if (fieldEl.getAttribute("name") === "email_to") {
+                        // We need this to be able to display what is the
+                        // dynamic value (the company email) in edit mode.
+                        fieldEl.dataset.dynamicValueNoSave = fieldEl.getAttribute("value");
+                    }
```

Comment:
Nah, we cannot rely on the fact the edit mode will be launched after this `destroy` call. Especially since you seem to write on something but never removing it... without testing, I guess this is saved as `data-dynamic-value-no-save` once the page is saved afterwards :grimacing: 

I guess the proper generic way would be for this "auto-fill of default values" part of this public widget to also be available in edit mode (thanks to `disabledInEditableMode: false`) but I don't think this is what we want functionally either: for other form fields, the auto-fill values should probably not be shown otherwise it is confusing (e.g. I choose to put "Anonymous" as default value for the "Name", I don't want "Mitchel Admin" hiding what I am choosing as default value).

So this seems to be a behavior we want: in edit mode only, for a specific field only (email_to). It is actually already an exception for the auto-fill system it seems:
```javascript
                    // In general, we want the data-for and prefill values to
                    // take priority over set default values. The 'email_to'
                    // field is however treated as an exception at the moment
                    // so that values set by users are always used.
                    if (name === 'email_to' && fieldEl.value
                            // The following value is the default value that
                            // is set if the form is edited in any way. (see the
                            // website.form_editor_registry module in editor
                            // assets bundle).
                            // TODO that value should probably never be forced
                            // unless explicitely manipulated by the user or on
                            // custom form addition but that seems risky to
                            // change as a stable fix.
                            && fieldEl.value !== 'info@yourcompany.example.com') {
                        continue;
                    }
```
=> Let's make it an exception option-wise too. In the related, `_computeWidgetState`: "if email_to has a value, show it, otherwise fallback on the data-for if any, otherwise show info@yourcompany.example.com (apparently) ?", without relying on what the readonly mode public widget does.